### PR TITLE
feat(job-details-modal): Added modal for viewing information for job run

### PR DESCRIPTION
**Pull request by jatinriverlane**
## Changes
* Add component for modal for job details
* Minor refactoring

### CODE REVIEW

**Review by samiralavi — APPROVED** [review]
All looks good to me, good work @jatinriverlane

File: aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { useState } from "react";
 
 import { JobDataType, JobsListColumnsType } from "types/globalTypes";
 import { jobListRowsPerPageOptions } from "constants/constants";
+import JobDetailsModal from "../JobDetailsModal";
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx:12]:
```suggestion
import JobDetailsModal from "components/organisms/JobDetailsModal";
```

File: aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx [hunk 2]
```diff
@@ -69,8 +72,7 @@ function JobsListTable({
                                     }}
                                     onMouseLeave={() => setShowActionId("-1")}
                                     onClick={() =>
-                                        // !TODO: after TT-122 is completed
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx:72]: Let's still keep this one comment

File: aqueductcore/frontend/src/components/organisms/JobDetailsModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+
+const settingItems = [
+    {
+        id: "parameters",
+        title: "Parameters",
+        component: <ActionParameters 
+                readonly
+                parameters={parameters}
+                inputParams={inputParams}
+                setInputParams={() => {}}
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/JobDetailsModal/index.tsx:133]: This line is extra.
We can make the `ActionParameters/>` component props conditional using [this approach](https://stackoverflow.com/a/75433249/10597542).
We only need to have `setInputParams ` if it's not readonly.

File: aqueductcore/frontend/src/components/molecules/LogsViewer/index.tsx
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/molecules/LogsViewer/index.tsx:1]: Sorry for being pedantic.
I feel like this is a `LogViewer` actually not _logs_. I might be wrong as a non-native speaker, but my understanding it that logs is a list the includes several log, but here we just do log to single run.

File: aqueductcore/frontend/src/components/molecules/LogsViewer/index.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+const LogsViewerBox = styled(Box)`
+    width: 100%;
+    height: 100%;
+    padding: ${(props) => props.theme.spacing(0.5)} ${(props) => props.theme.spacing(1.5)};
+`;
+
+const LogText = styled("pre")`
+    font-size: 0.813rem;
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/molecules/LogsViewer/index.tsx:14]: As a general comment, I guess these `font-size` are coming from Figma? let's ignore it with _this precision_, if it's really annoying in the design we can do `0.8` maybe? or the better approach is to use font-size as css-var.

File: aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx [hunk 5]
```diff
@@ -54,6 +53,12 @@ const HeaderIcon = styled(AutoAwesomeIcon)`
     vertical-align: middle;
 `;
 
+const CloseModalIcon = styled(CloseIcon)`
+    cursor: pointer;
+    line-height: 3.313rem;
```
Comment (safeamiiir) [inline aqueductcore/frontend/src/components/organisms/ExtensionModal/index.tsx:58]: same comment as i mentioned for `font-size` this precision is a bit unnecessary IMO.

**Review by safeamiiir — CHANGES_REQUESTED** [review]

File: aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx (hunk 2 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx:72]: Sure, adding this line again

File: aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx (hunk 1 above)
Comment (jatinriverlane) [reply aqueductcore/frontend/src/components/organisms/JobsListTable/index.tsx:12]: Done, updated

**Review by safeamiiir — APPROVED** [review]

**Pull request merged**